src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c

```diff
@@ -21,30 +43,242 @@ const ActorInit Obj_Raillift_InitVars = {
     (ActorFunc)ObjRaillift_Update,
     (ActorFunc)ObjRaillift_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19910.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static CollisionHeader* sColHeaders[] = { &D_06004FF8, &D_060048D0 };
+
+
+void ObjRaillift_UpdatePosition(ObjRaillift* this, s32 idx) {
+    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->points[idx]);
+}
+
+void ObjRaillift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRaillift* this = THIS;
+    s32 pad;
+    Path* path;
+    s32 type = OBJRAILLIFT_GET_TYPE(thisx);
+    s32 isColorful = false;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+
+    thisx->shape.rot.x = 0;
+    thisx->world.rot.x = 0;
+    thisx->shape.rot.z = 0;
+    thisx->world.rot.z = 0;
+    BcCheck3_BgActorInit(&this->dyna, 1);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, sColHeaders[type]);
+    this->speed = OBJRAILLIFT_GET_SPEED(thisx);
+    if (this->speed < 0.0f) {
+        this->speed = -this->speed;
+        isColorful = true;
+    }
+    if (type == DEKU_FLOWER_PLATFORM) {
+        Actor_SpawnWithParent(&globalCtx->actorCtx, thisx, globalCtx, ACTOR_OBJ_ETCETERA, thisx->world.pos.x,
+                              thisx->world.pos.y, thisx->world.pos.z, thisx->shape.rot.x, thisx->shape.rot.y,
+                              thisx->shape.rot.z, 0);
+        if (isColorful) {
+            thisx->draw = ObjRaillift_DrawDekuFlowerPlatformColorful;
+        } else {
+            thisx->draw = ObjRaillift_DrawDekuFlowerPlatform;
+        }
+    }
+    if (this->speed < 0.01f) {
+        this->actionFunc = ObjRaillift_DoNothing;
+    } else {
+        path = &globalCtx->setupPathList[OBJRAILLIFT_GET_PATH(thisx)];
+        this->curPoint = OBJRAILLIFT_GET_STARTING_POINT(thisx);
+        this->endPoint = path->count - 1;
+        this->direction = 1;
+        this->points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        ObjRaillift_UpdatePosition(this, this->curPoint);
+        if (OBJRAILLIFT_HAS_FLAG(thisx) && !Flags_GetSwitch(globalCtx, OBJRAILLIFT_GET_FLAG(thisx))) {
+            this->actionFunc = ObjRaillift_Idle;
+        } else {
+            this->actionFunc = ObjRaillift_Move;
+        }
+    }
+}
+
+void ObjRaillift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRaillift* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjRaillift_DoNothing(ObjRaillift* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion
void ObjRaillift_Noop(ObjRaillift* this, GlobalContext* globalCtx) {
```
I think `Noop` is convention for if a function doesn't do anything

Reply:
I've seen `DoNothing` used when it's an action function that doesn't do anything.

Reply:
Either should be fine then :)